resources/rest-service/cloudify/triggers/execute_workflow.clj

```diff
@@ -2,24 +2,37 @@
   [ctx]
     (let [deployment-id         (:deployment_id ctx)
           parameters            (:trigger-parameters ctx)
+          rest-token            "{{rest_api_token}}"
           rest-host             (or (System/getenv "REST_HOST") "127.0.0.1")
           rest-protocol         "https"
-          raw-rest-port         (or (System/getenv "REST_PORT") "80")
+          raw-rest-port         (or (System/getenv "REST_PORT") "53333")
           rest-port             (Integer/parseInt raw-rest-port)
-          verify-ssl-cert       false
-          base-uri              (str rest-protocol "://" rest-host ":" rest-port "/api/v2")
+          tenant-id             "{{tenant_id}}"
+          base-uri              (str rest-protocol "://" rest-host ":" rest-port "/api/v3")
           endpoint              (str "/executions")
           resource-uri          (str base-uri endpoint)
           body                  (cheshire.core/generate-string {
                                   :deployment_id           deployment-id
                                   :workflow_id             (:workflow parameters)
                                   :force                   (:force parameters)
                                   :allow_custom_parameters (:allow_custom_parameters parameters)
-                                  :parameters              (:workflow_parameters parameters)})]
-      (clj-http.client/post resource-uri
-        {:content-type   :json
-         :accept         :json
-         :socket-timeout (:socket_timeout parameters)
-         :conn-timeout   (:conn_timeout parameters)
-         :insecure?      (not verify-ssl-cert)
-         :body           body})))
+                                  :parameters              (:workflow_parameters parameters)})
+          execute-api-call      (fn [] (clj-http.client/post resource-uri
+                        {:content-type   :json
+                         :accept         :json
+                         :socket-timeout (:socket_timeout parameters)
+                         :conn-timeout   (:conn_timeout parameters)
+                         :headers {"API-Authentication-Token" rest-token
+                                   "Tenant" tenant-id}
+                         :trust-store "/etc/cloudify/ssl/cloudify_internal.p12"
+                         :trust-store-type "pkcs12"
+                         :trust-store-pass "cloudify"
+                         :body           body}))]
+    (try
+        (execute-api-call)
+        (catch Exception e 
+             (error "Exception calling Cloudify API: " e)
```

Comment:
Hmm. It probably should, yes.

I can try to figure out how to get it to use the pub-event, or (possibly a better plan?) I could modify publish-policy-error to have it pub-event as well as pub-log. Because it's attached to the exceptions stream it should then publish any errors that occur during policy processing. Unsure if that's the right thing to do or not...